config/config.go

```diff
@@ -88,7 +107,10 @@ func (c Config) Save(path string) (err error) {
 // 3.  The value of $XDG_CONFIG_PATH/func if the environment variable exists.
 // The path is created if it does not already exist.
 func Path() (path string) {
-	path = DefaultConfigPath
+	// default path is a relative path used in the unlikely event that
+	// the user has no home directory (no ~), there is no
+	// XDG_CONFIG_HOME set
+	path = filepath.Join(".config", "func")
 
 	// ~/.config/func is the default if ~ can be expanded
 	if home, err := homedir.Expand("~"); err == nil {
```

Comment:
I wonder, why do we need this sophisticated method? Wouldn't `os.UserHomeDir()` suffice?

Reply:
Was there problem in some containerised contexts?

Reply:
os.UserHomDir sounds like a better choice to me as well.  Changed.